src/btree/bt_vrfy.c

```diff
@@ -434,7 +434,8 @@ __wt_verify(WT_SESSION_IMPL *session, const char *cfg[])
             memset(&addr_unpack, 0, sizeof(addr_unpack));
             addr_unpack.newest_start_durable_ts = ckpt->start_durable_ts;
             addr_unpack.oldest_start_ts = ckpt->oldest_start_ts;
-            addr_unpack.oldest_start_txn = ckpt->oldest_start_txn;
+            addr_unpack.oldest_start_txn =
+              ckpt->write_gen > S2C(session)->base_write_gen ? ckpt->oldest_start_txn : 0;
```

Comment:
So when the checkpoint gets written out is it write gen equal to the connections write gen + 1?

Reply:
In `txn/txn_ckpt.c`, the function `__wt_checkpoint_tree_reconcile_update` sets the write gen of the checkpoint equal to the write gen of the btree.

In `meta/meta_ckpt.c`, the function `__wt_metadata_init_base_write_gen` initializes the base write gen of the connection to 1, then updates it as follows:

    conn->base_write_gen = WT_MAX(ckpt.write_gen + 1, conn->base_write_gen);